engine/table/src/main/java/io/deephaven/engine/table/impl/sources/ring/ObjectRingChunkSource.java

```diff
@@ -0,0 +1,44 @@
+package io.deephaven.engine.table.impl.sources.ring;
+
+import io.deephaven.chunk.ChunkType;
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.attributes.Values;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+
+// todo: this is not being auto-generated ATM
+final class ObjectRingChunkSource<T> extends AbstractRingChunkSource<T, Object[], ObjectRingChunkSource<T>> {
+    public static <T> RingColumnSource<T> columnSource(Class<T> type, int capacity) {
+        return new RingColumnSource<>(type, new ObjectRingChunkSource<>(type, capacity), new ObjectRingChunkSource<>(type, capacity));
+    }
+
+    public ObjectRingChunkSource(Class<T> type, int capacity) {
+        super(type, capacity);
+    }
+
+    @Override
+    public ChunkType getChunkType() {
+        return ChunkType.Object;
+    }
+
+    @Override
+    void clear() {
+        // todo: when do we call this?
+        Arrays.fill(ring, null);
+    }
+
+    @Override
```

Comment:
So, the abstract base class contains the full logic for `fillChunk`, and I don't think it's much more (if at all) inefficient than per-type ones would be (at least once JITd). And it's a lot less code. Also, `getChunk`. There is likely JIT advantages as well since all of the math-code is exactly the same; and if you break it out into per-type code, your code size and paths are larger.

Reply:
I've pushed down the fillers.